edi_storage/models/edi_backend.py

```diff
@@ -76,3 +81,47 @@ def _component_sort_key(self, component_class):
         return (
             1 if getattr(component_class, "_storage_backend_type", False) else 0,
         ) + res
+
+    def _cron_check_storage_pending_input(self, **kw):
+        for backend in self.filtered(lambda b: b.storage_id):
+            backend._check_storage_pending_input(**kw)
+
+    def _check_storage_pending_input(self, **kw):
+        self.ensure_one()
+        if not self.storage_id or not self.input_dir_pending:
+            _logger.info(
+                "%s ignored. No storage and/or input directory specified.", self.name
+            )
+            return False
+
+        pending_files = self.storage_id.list_files(self.input_dir_pending)
+        exchange_type = self.env["edi.exchange.type"].search(
+            self._domain_exchange_type_storage_pending_input()
+        )
+        if not len(exchange_type) == 1:
+            _logger.info("%s ignored. More than one exchange type found.", self.name)
+            return False
+        for file_name in pending_files:
+            existing = self.env["edi.exchange.record"].search(
+                [
+                    ("backend_id", "=", self.id),
+                    ("type_id", "=", exchange_type.id),
+                    ("exchange_filename", "=", file_name),
+                ]
+            )
+            if existing:
+                continue
+            self.create_record(
+                exchange_type.code, self._get_exchange_record_vals(file_name)
+            )
+            _logger.debug("%s: new exchange record generated.", self.name)
+        return True
```

Comment:
I'm not sure all this logic should stay here :thinking: 
I would be more in favor to use a specific component for this machinery.
Here we should only have the cron hook looking for exchange types to work with.
Also, consider that a specific backend type or exchange type could require a different logic.
Using a component would make this easier.

Reply:
you have to define at least a basic process. The same way it is done here: https://github.com/OCA/edi/blob/13.0/edi/models/edi_backend.py#L314. What can be different in this other than the storage used, which is already abstracted in the specific modules (I only use the high level method `list_files`).

If you still think this need to be moved to a component, can you help me with naming/structure? can you propose me names for these components?

Reply:
fine, let's keep it simple. I opened this https://github.com/ForgeFlow/edi/pull/1 with some improvements (use file pattern, prefix all methods w/ `_storage`, use jobs, well... you'll see).
Let me know what you think!

Reply:
Will review now, I just pushed some changes, hope we don't conflict :crossed_fingers: